libevmasm/CommonSubexpressionEliminator.h

```diff
@@ -179,7 +179,12 @@ AssemblyItemIterator CommonSubexpressionEliminator::feedItems(
 )
 {
 	assertThrow(!m_breakingItem, OptimizerException, "Invalid use of CommonSubexpressionEliminator.");
-	for (; _iterator != _end && !SemanticInformation::breaksCSEAnalysisBlock(*_iterator, _msizeImportant); ++_iterator)
+	unsigned const maxChunkSize = 1000;
+	for (
+		unsigned chunkSize = 0;
+		_iterator != _end && !SemanticInformation::breaksCSEAnalysisBlock(*_iterator, _msizeImportant) && chunkSize < maxChunkSize;
+		++_iterator, ++chunkSize
+	)
 		feedItem(*_iterator);
 	if (_iterator != _end)
```

Comment:
I think this should be `_iterator != _end && chunkSize < maxChunkSize`.

Reply:
You are correct. Fixup commit added.